app/src/main/java/com/example/internetshop/presentation/viewModel/ProductsListViewModel.kt

```diff
@@ -20,7 +21,7 @@ class ProductsListViewModel @Inject constructor(
 
     fun getCategoryProductList(categoryName: String) {
         compositeDisposable.add(
-            fromCategoryUseCase.execute(categoryName)
+            fromCategoryUseCase.execute(CategoryRequest(categoryName))
```

Comment:
You have passed CategoryRequest into domain use case. Category request it's data model. Domain should you it own domain model. Also inside CategoryRequest class in init block you have `it.tolowercase` please use mappers!!

Reply:
fixed